### PR TITLE
Include WIT bindings

### PR DESCRIPTION
**Pull request by Michael-F-Bryan**
This PR updates the `wabt` embeddings to be in sync with the latest version of `wit-bindgen` ([c047230](https://github.com/wasmerio/wit-bindgen/commit/c04723063c7a5a7389660ca97f85ffd9bc9ef0b8)). I've also included the `wapm.toml` and `README-wapm.md` that is present [on WAPM](https://wapm.io/syrusakbary/wabt).

### CODE REVIEW

**Comment by Michael-F-Bryan** [comment]
The `wapm validate` command is currently failing because of an `InterfaceViolated` error... Presumably WASI has changed since the `wabt` package was released 3 years ago, and now the `interfaces = {wasi= "0.0.0-unstable"}` line under each `[[module]]` in `wapm.toml` needs to be updated.

```console
$ wapm validate .
Error: WASM file "./out/wasi/Debug/wasm-interp" detected as invalid because InterfaceViolated { 
  errors: [
    "Missing import \"wasi_snapshot_preview1\" \"fd_close\"", 
    "Missing import \"wasi_snapshot_preview1\" \"fd_seek\"", 
    "Missing import \"wasi_snapshot_preview1\" \"fd_write\""
  ] 
}
```

**Comment by Michael-F-Bryan** [comment]
@syrusakbary can you have a look through this PR to see if I'm doing things right? 

I've noticed that running `make` from the root directory doesn't generate the same folder structure as the `wabt` package on WAPM. Do I need to do something extra to generate a release build?

File: wapm.toml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+name = "wasm-interp"
+module = "wasm-interp"
+
+[[command]]
+name = "wasm-validate"
+module = "wasm-validate"
+
+[[command]]
+name = "wasm-strip"
+module = "wasm-strip"
```
Comment (syrusakbary) [inline wapm.toml:74]: We probably want to add the wit-bindings as well?

**Comment by syrusakbary** [comment]
Things look more or less right (to generate libwabt).

I think that to generate the rest of WASI CLI files we need to use cmake (so we need to have that working)

File: wapm.toml (hunk 1 above)
Comment (Michael-F-Bryan) [reply wapm.toml:74]: I thought commands don't have bindings because they're intended as executables that people run, not libraries that people import?

**Comment by Michael-F-Bryan** [comment]
> I think that to generate the rest of WASI CLI files we need to use cmake (so we need to have that working)

Do you mean the executables? The `wapm.toml` already includes commands for all the executables that you can use at the moment, so I'd say that any further changes to the `wabt` repo (e.g. modifying `CMakeLists.txt` to generate new files or build things differently) are out of scope for this project.

File: wapm.toml (hunk 1 above)
Comment (syrusakbary) [reply wapm.toml:74]: Packages can have both commands and libraries.

That means, that a package can have 3 CLI binaries (with WASI), and 1 library (with WIT), for example.

So basically, we can have a new `libwabt` module that has bindings in the package :)

Comment (Michael-F-Bryan) [reply wapm.toml:74]: Oh, that was [already added](https://github.com/Michael-F-Bryan/wabt/blob/ea60d90ca11f3763bb3737a7e7b6a06580f61c11/wapm.toml#L8-L13) yesterday before marking the PR ready. I'm guessing I forgot to push the code up.

**Review by syrusakbary — APPROVED** [review]

**Pull request merged**